File: ICObenchAPI.js
const request = require('request');
const CryptoJS = require('crypto-js');

class ICObench {
    constructor(){
        this.publicKey = "";
        this.privateKey = "";
        this.apiUrl = 'https://icobench.com/api/v1/';
    }

    get(action, callback, data = {}) {
        let dataJSON = JSON.stringify(data);

        let sign = CryptoJS.HmacSHA384(dataJSON, this.privateKey);
        sign = CryptoJS.enc.Base64.stringify(sign);

        request.post(this.apiUrl + action, {
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'X-ICObench-Key': this.publicKey,
                'X-ICObench-Sig': sign,
            },
            forever: true,
            json: data
        }, function (error, response, body){
            console.log(body);
        });
    }
}

let test = new ICObench();

test.get("icos/trending", (data) => {console.log(data)});

File: python/ICObenchAPI.py
import hmac
import json
import hashlib
import base64
import urllib2

publicKey = ""
privateKey = ""

apiUrl = 'https://icobench.com/api/v1/'

data = {'search': 'DMarket'}

hash = hmac.new(privateKey, '', hashlib.sha384)

dataJSON = json.dumps(data)
hash.update(dataJSON)
sign = hash.digest()
sign = base64.b64encode(sign)

request_headers = {
'Accept': 'application/json',
'Content-Type': 'application/json',
'X-ICObench-Key': publicKey,
'X-ICObench-Sig': sign
}

print apiUrl + "icos/all"

request = urllib2.Request(apiUrl + "icos/all", data=dataJSON, headers=request_headers)
contents = urllib2.urlopen(request).read()
print contents
